addons/crm/models/crm_lead.py

```diff
@@ -982,19 +991,6 @@ def _stage_find(self, team_id=False, domain=None, order='sequence, id', limit=1)
     # ACTIONS
     # ------------------------------------------------------------
 
-    def toggle_active(self):
```

Comment:
That was done on purpose, separating the functional click from automatic archive. For CRM please keep the old behavior. Purpose is to keep tests untouched.

Reply:
In current implementation, if someone writes on the active field, it won't recompute the probability or lost reason which will lead to inconsistencies.
The updated tests only calls the action to archive/unarchive based on the active value instead of going though a toggle.

Could you confirm that I should overwrite `action_archive` and `action_unarchive` instead of `write`?

Reply:
Yep, because we differentiate functional flows (using the "archive" in the UI) from the technical flow (writing on active field). For some reason you can archive leads or opportunities, and if you do that using low level write, it should not impact your PLS, it is just some cleaning you do. So here action_archive does more than just "write active False" and that is intended :)

Reply:
Ok, I will adapt.
Still, I find it strange because I though that active=False implied that the lead was lost. For example, `_track_subtype` indicates that the lead is lost.

Reply:
Yep there are probably some inconsistencies, will check :) but the idea was (and normally still is) that you can write active=False on a lead or an opportunity manually without telling it is lost, there are a lot of reasons why you want to archive something and to hide it (maybe it is a duplicate for example). Marking as lost is a business choice, you lost the opportunity and you won't make the deal.

Thanks for updating !